Kotlin: Add the Sagemaker pipeline workflow (#5230)
awsdocs/aws-doc-sdk-examples: kotlin/usecases/workflow_sagemaker_pipes/Readme.md

### Geospatial jobs
A SageMaker pipeline can be used for model training, setup, testing, or validation. This example uses a simple job
for demonstration purposes: a [Vector Enrichment Job (VEJ)](https://docs.aws.amazon.com/sagemaker/latest/dg/geospatial-vej.html) that processes a set of coordinates to produce human-readable 
addresses powered by Amazon Location Service. Other types of jobs could be substituted in the pipeline instead.

## ⚠ Important

* Running this code might result in charges to your AWS account.
* Running the tests might result in charges to your AWS account.
* We recommend that you grant your code least privilege. At most, grant only the minimum permissions required to perform the task. For more information, see [Grant least privilege](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html#grant-least-privilege).
* This code is not tested in every AWS Region. For more information, see [AWS Regional Services](https://aws.amazon.com/about-aws/global-infrastructure/regional-product-services).

## Scenario

### Prerequisites

To use this tutorial, you need the following:

+ An AWS account

> + A Java IDE (this tutorial uses the IntelliJ IDE)

Author: Fixed

+ Java JDK 1.8
+ Gradle 6.8 or higher
+ Setup your development environment. For more information, see [Setting up the AWS SDK for Kotlin](https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html).

To view pipelines in SageMaker Studio, you need to [set up an Amazon SageMaker Domain](https://docs.aws.amazon.com/sagemaker/latest/dg/gs-studio-onboard.html).

To use geospatial capabilities, [you need to use a supported Region](https://docs.aws.amazon.com/sagemaker/latest/dg/geospatial.html).

You must download and use these files to successfully run this code example:

+ GeoSpatialPipeline.json
+ latlongtest.csv

> These files are located on GitHub in [AWS SDK for .NET](https://github.com/awsdocs/aws-doc-sdk-examples/tree/main/dotnetv3/SageMaker/Scenarios).

Reviewer: This .NET link looks odd, if you are going to use the exact same files I think they should go in the Resources folder, for both Kotlin and Java.

Author: I do not think this is an issue. Just a place where the files are located.

Reviewer: I disagree, it looks odd to have a link that looks like it goes to another SDK (even though it is an example file). The workflow gives the option to use different files, but if you're using the exact same ones it should be in a shared location.

### Kotlin Lambda Function

To successfully run this example, you need to create the Kotlin Sagemaker Lambda function by creating a .jar file and then placing the .jar file into an Amazon S3 bucket. You can find this project here: [Create the SageMaker geospatial Lambda function using the AWS SDK for Kotlin](https://github.com/awsdocs/aws-doc-sdk-examples/tree/main/kotlin/usecases/workflow_sagemaker_lambda).  

Once you create the Kotlin Lambda project, you can build the required .jar file by using the shadowJar plugin .

> ![AWS Tracking Application](images/shawdow.png)

Reviewer: Why is this image called "AWS Tracking application"?

Perform the following tasks:

1. Create a new Kotlin project using an IDE and import the project from Github. 
2. Create the FAT JAR by running shadowJar.
3. Use the AWS Management Console to place the FAT JAR into the Amazon S3 bucket. This S3 bucket name is input to this Kotlin program. 
 

### Instructions

> You can run this Java code example from within your Java IDE.

Reviewer: Again, references Java instead of Kotlin.

* Running the tests might result in charges to your AWS account.
* We recommend that you grant your code least privilege. At most, grant only the minimum permissions required to perform the task. For more information, see [Grant least privilege](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html#grant-least-privilege).
* This code is not tested in every AWS Region. For more information, see [AWS Regional Services](https://aws.amazon.com/about-aws/global-infrastructure/regional-product-services).

## Scenario

### Prerequisites

To use this tutorial, you need the following:

+ An AWS account
+ A Java IDE (this tutorial uses the IntelliJ IDE)
+ Java JDK 1.8
+ Gradle 6.8 or higher
+ Setup your development environment. For more information, see [Setting up the AWS SDK for Kotlin](https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html).

To view pipelines in SageMaker Studio, you need to [set up an Amazon SageMaker Domain](https://docs.aws.amazon.com/sagemaker/latest/dg/gs-studio-onboard.html).

To use geospatial capabilities, [you need to use a supported Region](https://docs.aws.amazon.com/sagemaker/latest/dg/geospatial.html).

> You must download and use these files to successfully run this code example:

Reviewer 2: ```suggestion
To successfully run this code example, you must download and use the following files:
```

+ GeoSpatialPipeline.json
+ latlongtest.csv

> These files are located on GitHub in this folder [sample_files](https://github.com/awsdocs/aws-doc-sdk-examples/tree/main/resources/sample_files).

Reviewer 2: ```suggestion
These files are located on GitHub in the [sample_files](https://github.com/awsdocs/aws-doc-sdk-examples/tree/main/resources/sample_files) folder.
```

> ### Kotlin Lambda Function

Reviewer 2: ```suggestion
### Kotlin Lambda function
```
